Sources/SignalProducer.swift

```diff
@@ -464,6 +464,16 @@ extension SignalProducerProtocol {
 		return lift { $0.filter(predicate) }
 	}
 
+	/// Applies `transform` to values from the producer and forwards values with non `nil` results unwrapped.
+	/// - parameters:
+	///   - transform: A closure that accepts a value from the `value` event and
+	///                returns a new optional value.
+	///
+	/// - returns: A producer that will send new values, that are non `nil` after the transformation.
+	public func filterMap<U>(_ transform: @escaping (Value) -> U?) -> SignalProducer<U, Error> {
```

Comment:
The `Sequence.flatMap` overload you're referring to (which removes `nil` values after transformation) is a weird one: it doesn't behave as `flatMap` should according to general FP laws/expectations, and it should probably be renamed `compactMap`. While I see creating parity with the standard library appealing, I'd rather keep things a bit more predictable within the functional expectations of FRP.